examples/mugen/retrieval/video_clip.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import nn
+from torchmultimodal.utils.common import get_current_device
+from transformers import DistilBertConfig, DistilBertModel, DistilBertTokenizer
+
+
+class TextEncoder(nn.Module):
+    """Encode text to a fixed-length vector
+
+    Taken from MUGEN's text encoder
+        (https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/videoclip/modules.py)
+    """
+
+    def __init__(
+        self,
+        model_name="distilbert-base-uncased",
+        pretrained=True,
+        max_length=200,
+    ):
+        super().__init__()
+        self.max_length = max_length
+        self.tokenizer = DistilBertTokenizer.from_pretrained(model_name)
+        if pretrained:
+            self.model = DistilBertModel.from_pretrained(model_name)
+        else:
+            self.model = DistilBertModel(config=DistilBertConfig())
```

Comment:
I can have the constructor accept a `model_config` dict and then pass in `DistilBertConfig(**model_config)` here

Reply:
I would bias more towards giving access to specific parameters over configs since blindly passing configs around can make things less interpretable. But I understand that the config approach is better aligned with this HF model. So ultimately I think it's up to you which you prefer 🙂

Reply:
I would prefer the config approach since none of the parameters strike me as more important than the others and there are so many, but maybe having the user pass in a DistilBertConfig object (rather than a dict) adds some transparency to what params are configurable?